src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -375,4 +375,92 @@ int32_t GlobalizationNative_GetSortKeyNative(const uint16_t* localeName, int32_t
     }
 }
 
+int32_t GlobalizationNative_GetUIUnicodeVersion(const uint16_t* localeName, int32_t localeNameLength)
+{
+    @autoreleasepool {
+        // This function mimics ICU's ucol_getVersion which returns a collator version
+        // ucol_getVersion can return different versions for different locale collations
+        // Format: UVersionInfo[4] which is a 4-byte array packed into an int32_t
+        
+        NSOperatingSystemVersion osVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+        
+        // In ICU, collator versions typically follow this format:
+        // - Byte 0: Major version (related to Unicode version)
+        // - Byte 1: Minor version (related to Unicode version)
+        // - Byte 2: Milli version (for tailorings or collator-specific changes)
+        // - Byte 3: Micro version (build number or additional distinction)
+        
+        // Calculate major version based on OS version
+        // iOS versions and macOS versions are not aligned (e.g., iOS 18 and macOS 15 can coexist)
+        // so we need to handle them differently
+        uint8_t collatorMajor;
+        
+        // A simple but effective approach to differentiate iOS/macOS:
+        // iOS uses higher version numbers (e.g., 17+) while macOS uses lower numbers (e.g., 13-14)
+        // This approximation won't be perfect for all future versions but gives consistent behavior
+        if (osVersion.majorVersion >= 10) {
+            // Likely iOS/iPadOS - Apply iOS formula
+            // Approximation: iOS 17 -> Unicode 14, iOS 16 -> Unicode 13, etc.
+            collatorMajor = (uint8_t)(osVersion.majorVersion - 3);
+        } else {
+            // Likely macOS - Apply macOS formula
+            // macOS 14 (Sequoia) -> Unicode 15, macOS 13 (Ventura) -> Unicode 14, etc.
+            collatorMajor = (uint8_t)(osVersion.majorVersion + 1);
+        }
+
+        uint8_t collatorMinor = 0;
+        uint8_t collatorMilli = 0;
+        uint8_t collatorMicro = 0;
+        
+        // Get the NSLocale from the provided locale name
+        NSLocale *currentLocale = GetCurrentLocale(localeName, localeNameLength);
+        NSString *localeIdentifier = [currentLocale localeIdentifier];
+        
+        // The milli version can be used to distinguish between Apple platform variations
+        // Use a value that represents the platform type
+        // This helps ensure consistent collation behavior across releases while differentiating platforms
```

Comment:
Comment grammar is all out of whack, smh

```suggestion
        // The milli version can be used to distinguish between Apple platform variations.
        //
        // Use a value that represents the platform type. This helps ensure consistent collation behavior across releases while differentiating platforms.
```

This version should have the proper paragraph break, and the text wrap should occur automatically when you print the book.